wasi/wasi_test.go

```diff
@@ -312,7 +312,69 @@ func TestAPI_ClockTimeGet_Errors(t *testing.T) {
 // TODO: TestAPI_ProcExit TestAPI_ProcExit_Errors
 // TODO: TestAPI_ProcRaise TestAPI_ProcRaise_Errors
 // TODO: TestAPI_SchedYield TestAPI_SchedYield_Errors
-// TODO: TestAPI_RandomGet TestAPI_RandomGet_Errors
+
+// randomWat is a wasm module to call random_get.
+//go:embed testdata/random.wat
+var randomWat []byte
+
+func TestAPI_RandomGet(t *testing.T) {
+	store, wasiAPI := instantiateWasmStore(t, randomWat, "test")
+	maskLength := 7 // number of bytes to write '?' to tell what we've written
+	expectedMemory := []byte{
+		'?',                          // random bytes in `buf` is after this
+		0x53, 0x8c, 0x7f, 0x96, 0xb1, // random data from seed value of 42
+		'?', // stopped after encoding
+	} // tr
+
+	var buf_len = uint32(5) // arbitrary buffer size,
+	var buf = uint32(1)     // offset,
+	var seed = int64(42)    // and seed value
+
+	t.Run("API.RandomGet", func(t *testing.T) {
+		maskMemory(store, maskLength)
+		// provide a host context with a seed value for random generator
+		hContext := wasm.NewHostFunctionCallContextWithSeed(context.Background(), store.Memories[0], seed)
+
+		errno := wasiAPI.RandomGet(hContext, buf, buf_len)
```

Comment:
maybe just change this all over from buf_len -> bufLen except in the %.wat

Reply:
Thanks @codefromthecrypt 
Got it, will use bufLen overall (except %.wat)